qiskit_ibm_provider/ibm_backend.py

```diff
@@ -796,7 +796,9 @@ def _deprecate_id_instruction(
 
     def get_translation_stage_plugin(self) -> str:
         """Return the default translation stage plugin name for IBM backends."""
-        return "ibm_dynamic_circuits"
+        if "id" not in self.target:
+            return "ibm_dynamic_circuits"
+        return None
```

Comment:
I believe it would be better to instead conditionally add the pass [here](https://github.com/Qiskit/qiskit-ibm-provider/blob/68d4fcec036c1705be3c1fa832063e65bb992181/qiskit_ibm_provider/transpiler/plugin.py#L82) as the way this is implemented right now might yield unexpected behaviour in the future if additional functionality is added to `ibm_dynamic_circuits` in addition to the `IDGate` -> `Delay` conversion

Reply:
Done in: https://github.com/Qiskit/qiskit-ibm-provider/pull/557/commits/48b032d83aea5c23a79342922064215386d01887